### ISSUE TITLE
Replace Seven tablesort icons with SVGs

### ISSUE THREAD

**Issue by robertgarrigos** [backdrop/backdrop-issues#5865]
As suggested [here](https://github.com/backdrop/backdrop-issues/issues/3906#issuecomment-1323636185), I'm posting this again as a new issue.

I just come across [this](https://github.com/backdrop/backdrop-issues/issues/3906) issue when I realized that the icons we are showing to sort a table now are inducing me to confusion.

Right now we are showing these icons:
![131394193-4a8addda-f2cc-4142-a2d0-f6f17be3e4bc](https://user-images.githubusercontent.com/105581/202693505-105ebf26-f215-48e0-a936-6dad51236f5d.png)
which in fact mean the same: ascendant sorting, from the smaller line to the larger line. The only difference is that there are shown upside-down.

While these icons:
![131393993-29222af1-a780-4a9f-8bda-2412557ffd79](https://user-images.githubusercontent.com/105581/202694137-f48bafb6-9f2b-43cc-b936-c7ef50176cf5.png)
are really showing different sorting: descendant on the left, from the larger line to the smaller line, and ascendant on the right, from the smaller line to the larger line.

Thus, I would suggest using the second set of icons, instead of the first one.

**Comment by yorkshire-pudding** [comment]
Wow! I've always found these confusing but never noticed that this is why. Yes please to fixing this.

**Comment by olafgrabienski** [comment]
As mentioned in https://github.com/backdrop/backdrop-issues/issues/3906#issuecomment-1320011866, the 2nd set is better when I look at the icons in an analytical way, but when I have a quick look at the icons, the first icon set seems more appropriate to me (easier to recognize the direction of the arrow). Looking at the [original issue thread](https://github.com/backdrop/backdrop-issues/issues/3906), some people preferred the second set, many others the first.

It looks like both icon sets are not ideal, as each of them seems to confuse its own audience. Maybe we should abstain from using the bars and just use arrows? That should be more clear for both groups.

**Comment by kiamlaluno** [comment]
To me, both the set of icons are confusing; it's the combination of arrow and bars that make the icons hard to understand.

Looking at the bars, the first icon of the first set suggests me an ascending order, while the second icon suggests me a descending order; the arrows would tell me "from the smaller to the bigger" in both the cases, and that is confusing for me, who looks at vertically placed objects from the top to the bottom.

Looking at the bars, both the icons of the second set suggest me the same order. The arrows are still confusing me, who still looks at vertically placed objects from the top to the bottom.

**Comment by jenlampton** [comment]
> To me, both the set of icons are confusing

I feel the same way. Should we consider going back to a simple carrot, or small arrow?

**Comment by olafgrabienski** [comment]
Yes please! Visually, I'd prefer the caret, but let's reconsider @klonos' objection that we use it also in fieldsets and drop buttons, even if there it stands for different actions. I don't see the different meanings as a problem. We could however bypass the controversy by using the arrow.

**Comment by yorkshire-pudding** [comment]
@olafgrabienski we discussed in the design meeting; the small arrow is less clear at small sizes than the caret.  Might be possible to mitigate if we can find a clearer icon.

**Comment by Mike-Ket** [comment]
While I like the icon itself, as a user, the two on the left make more sense to me. For both the rows I have to read them carefully (as opposed to scan-read them) in order to confirm what they are telling me.
I think this is because the icon is in two parts. The arrow says the column sorted. The scale bars show the direction of the sort. If a column isn't sorted no icon is needed, or perhaps an unsorted icon with no arrow and scale bars in random order.

**Comment by bugfolder** [comment]
Considering the combination of arrow and bar: an up-arrow tells me "the normal order of of this thing goes from bottom to top," and "the normal order" is clearly lexicographic for something called "Title". So an arrow alone is clear. The stack of bars with smallest bar on top tells me "the biggest thing is on the bottom, smallest thing is on the top," and that's a lot less clear for lexicographical ordering, because I don't think of "Z" as being "larger" or "heavier" than "A". It just comes later in normal ordering. So I agree that we should just go back to a caret or arrow, both of which indicate unambiguously that "the normal order of this thing goes this direction."

As for which we use, I definitely find the caret more readable at a distance or at smaller size.

All of this has been said in other words by other commenters above; just adding my agreement.

**Comment by klonos** [comment]
I'm fine with using just up/down arrows, but lets please not go back to using the caret, as that's being used for expand/collapse functionality in multiple places. Looking at https://fontawesome.com/search?q=arrow&o=r&m=free what do people prefer?

I personally like either the `arrow-up`/`arrow-down` or the `arrow-up-long`/`arrow-down-long` variants. Which set do you find easier to distinguish?:
- `arrow-up`/`arrow-down`:
  ![image](https://user-images.githubusercontent.com/2423362/221399057-f8644c7d-9a24-4f6d-96dd-bc18c548afe4.png)
- `arrow-up-long`/`arrow-down-long`:
  ![image](https://user-images.githubusercontent.com/2423362/221399095-c1aa044b-73bd-4e47-96c9-36f1b9d026c3.png)

**Comment by klonos** [comment]
`arrow-up`/`arrow-down`:
  ![image](https://user-images.githubusercontent.com/2423362/221399057-f8644c7d-9a24-4f6d-96dd-bc18c548afe4.png)

**Comment by klonos** [comment]
`arrow-up-long`/`arrow-down-long`:
  ![image](https://user-images.githubusercontent.com/2423362/221399095-c1aa044b-73bd-4e47-96c9-36f1b9d026c3.png)

**Comment by klonos** [comment]
...there's also a "bulky" `up-long`/`down-long` variant:
![image](https://user-images.githubusercontent.com/2423362/221399561-e75cbb7c-490a-41f5-affa-915d7e2d9373.png)

**Comment by kiamlaluno** [comment]
Is there any chance that people from different parts of the world could understand the arrow up differently?

**Comment by olafgrabienski** [comment]
> I'm fine with using just up/down arrows, but lets please not go back to using the caret, as that's being used for expand/collapse functionality in multiple places.

@klonos It's not necessarily problematic to use the same symbol for different things. In this case, I don't think it's problematic, and the caret is better readable, as many people expressed. So, I'd suggest to add the caret to the up/down-vote comparison.

**Comment by bugfolder** [comment]
I agree with @olafgrabienski that it's OK to use the same or similar symbols in different places when they're unlikely to be confused, but "arrow-up"/"arrow-down" are bold enough that they don't suffer from the same readability/distinguishability issue that small arrow (⬇️ ) did.

**Comment by yorkshire-pudding** [comment]
I think the first pair `arrow-up`/`arrow-down` are preferable to the other arrows.  I'm also happy with (and prefer) caret and do think it is more readable than the arrow.  Note also that font awesome has caret as:
[`sort-up`](https://fontawesome.com/icons/sort-up?s=solid&f=classic) / [`sort-down`](https://fontawesome.com/icons/sort-down?s=solid&f=classic)
So it is probably a well accepted pattern, and I also don't think it is a problem that it's also used for other things in different contexts.

**Comment by klonos** [comment]
> Is there any chance that people from different parts of the world could understand the arrow up differently?

Good question @kiamlaluno 🤔 ...I wasn't able to find anything related to language-specific representations of arrows, so unless anyone else finds any, I'm inclined to say that they are a universal thing.

**Comment by kiamlaluno** [comment]
@klonos I was thinking more of people who read text from the bottom to the top, who could understand the arrow differently. (I do not know if that even happens; that is why I asked.)

**Comment by stpaultim** [comment]
I am marking this as a possible Good First Issue, because there is some solid direction in the comments about which way to go and changing icons seems like it might be a good place for someone to start in core. Correct me if I'm wrong?

**Comment by olafgrabienski** [comment]
> (...) there is some solid direction in the comments about which way to go

If I understand the discussion correctly, most people prefer to change the sorting icons back to a caret (🔽).

**Comment by herbdool** [comment]
Related: https://github.com/backdrop/backdrop-issues/issues/6779 which is changing the icons in core to SVG. This one, I believe is about the icon in Seven.

**Comment by quicksketch** [comment]
We haven't had to deal with Phosphor renaming an icon, but that looks like that might happen in https://github.com/phosphor-icons/homepage/issues/508. If that were to occur, we can use `hook_icon_info()` to map the old name to the new path when we update the icon set.

So we're not strictly blocked on waiting for that to happen, but it will be confusing that our code will reference the icons opposite from the icon names.

Or as a completely different option, we could bundle the heavier "bold" variant of this SVG right into Seven, and then we wouldn't need to worry about waiting on the name at all, we could just rename it ourselves or specifically for Seven's use. I'll file a PR and see what that looks like.

**Comment by quicksketch** [comment]
PR filed at https://github.com/backdrop/backdrop/pull/4941 that keeps the original design intention of Seven's custom icons but switches them to use a custom icon (though this is actually just Phosphor's `sort-decending-bold` icon).

![image](https://github.com/user-attachments/assets/eae56d87-f5c7-4b61-9825-306dcc977713)

**Comment by quicksketch** [comment]
@NormPlum It is entirely likely I did the arrows backwards.

**Comment by quicksketch** [comment]
Well now having been told they're backwards I realize that -- to me -- the current situation makes sense. To me, an arrow pointing down means "top to bottom" or "first to last", and I would consider A to be first and Z to be last. I would also still consider this to be "ascending", despite the arrow pointing down. I know the arrow literally pointing down; I still find that more intuitive. I wonder how much this is a [blue/gold situation](https://en.wikipedia.org/wiki/The_dress). I'll defer to others as to which makes more sense.

**Comment by olafgrabienski** [comment]
I think an arrow pointing down is correct in this case, as the sort order is a result of the "Updated" column default order of the _Administer content_ view being descending.

I'm not sure how other systems handle sorting, but arrow up for ascending dates and arrow down for descending dates is also what Finder does in Mac OS.

**Comment by bugfolder** [comment]
Yeah, a down-arrow signifies to me also "first to last going down on the screen."

To get a little picky, these icons

![](https://user-images.githubusercontent.com/105581/202693505-105ebf26-f215-48e0-a936-6dad51236f5d.png)

seem to indicate "ordered from shortest titles to longest titles."

**Comment by stpaultim** [comment]
> I wonder how much this is a [blue/gold situation](https://en.wikipedia.org/wiki/The_dress). I'll defer to others as to which makes more sense.

@quicksketch Did you see the discussion that starts here and suggests that the solution to this might be to simply use arrows pointing up or down, with the bars alongside them. It seems as if the consensus is moving in that direction.

https://github.com/backdrop/backdrop-issues/issues/5865#issuecomment-1328706088

**Comment by avpaderno** [comment]
To confirm what @stpaultim said: To me, the sort icon shown in https://github.com/backdrop/backdrop-issues/issues/5865#issuecomment-2547409926 means *the bigger on the bottom, the smaller on the topé or *the older on the bottom, the newer on the top* because those bars have different lengths. That is why I think that without the bars the meaning is clearer.

**Comment by indigoxela** [comment]
This seems really tricky to fix. Last time we tried that in #3906, but didn't touch the wording of _titles_. The way ascending and descending are used (and have been used in D7, from where we inherited it), is really weird.

Here's what we currently have as default (_without_ the overrides by Seven, so displayed in Basis and all themes not overriding the defaults):

![sorting-confusion](https://github.com/user-attachments/assets/1534210b-d1eb-4a17-81f7-917a976948ef)

The new icon in #3906 was an attempt to improve the situation - but my suspicion is, that in the end we really made it even more confusing.

We could stick with an icon with bars (as in https://github.com/backdrop/backdrop/pull/4941), we could switch to arrows without bars, but both probably won't actually fix the confusion caused by the text (title). And I don't think, we can switch the text. Or could we? And would switching the title text even be an improvement, anyway? :stuck_out_tongue_winking_eye:

**Comment by quicksketch** [comment]
Since you point out the title text: it's something that bothers me: Is what's being shown here the current state? Or what will happen if I click on the arrow? We could clarify things quite a bit by indicating "currently sorted descending" (which would be the "alt" text) and "click to sort ascending" (which would be the "title" text") or maybe clarify with "Currently sorted descending, click to sort ascending" for both.

**Comment by indigoxela** [comment]
> We could clarify things quite a bit by indicating "currently sorted descending" (which would be the "alt" text) and "click to sort ascending" (which would be the "title" text")

Good idea. Currently we have an image with an unclear alt text and the surrounding link also has an unclear title text.
Using more explicit text might in fact improve things here. :thinking:

Does the icon show the current state or what happens when clicking? Personally, I'd expect it to show the _current_ status, but I might be totally wrong.

**Comment by herbdool** [comment]
I think the title is meant to be the action. Like "click to sort descending" but shorter. But, yes, that ends up being confusing.

**Comment by olafgrabienski** [comment]
> Does the icon show the current state or what happens when clicking?

It's also my understanding that while the icon shows the current state, the _alt_ and the _title_ attribute refer to the upcoming action. Making the attritbute texts clearer is an excellent idea, in my opinion.

**Comment by bugfolder** [comment]
> > Does the icon show the current state or what happens when clicking?
> 
> It's also my understanding that while the icon shows the current state, the _alt_ and the _title_ attribute refer to the upcoming action.

There seems to be no global consensus on on the "current state vs action" question, e.g., https://stackoverflow.com/questions/2716729/should-an-icon-show-current-state-or-next-state. Perhaps, though, if we settle on a policy here, we should capture it in the [docs site](https://docs.backdropcms.org) for future reference? There may be existing inconsistencies in contrib (or core) for which changing immediately might cause confusion, but at least having it documented would provide a target that UI implementations could evolve toward.

> Making the attritbute texts clearer is an excellent idea, in my opinion.

Ditto.

**Comment by indigoxela** [comment]
> There seems to be no global consensus on on the "current state vs action" question...

So, in our attempt to clarify (visual) expectations, we immediately fell into the next rabbit hole? :sob:

We could clarify the alt and title text a bit, though. Or is that also a trap? :stuck_out_tongue_winking_eye:

**Comment by avpaderno** [comment]
> We could clarify the alt and title text a bit, though. Or is that also a trap?

The tooltip for *Content type* says *Sort by Content type*. I guess the tooltip for the icon could say what happen when the icon is clicked. If that is not clear enough, *Click to* can be added and make the tooltip *Click to sort ascending* or *Click to sort descending*, which was suggested by @quicksketch. At least, *Click to sort ascending* would make clear it is not the current state.

**Comment by herbdool** [comment]
Here's my suggestion: that we focus on the icon and create a new issue for changing the wording, since we're getting stuck on that.

Currently, at least the arrow direction matches that in default theme function: the arrow points down when it's sorted descending (and the tooltip indicates the action: "sort ascending"). The PR:

<img width="729" height="136" alt="Image" src="https://github.com/user-attachments/assets/1d07ff51-f4b2-465d-821b-5ca886164fc9" />

In the default theme function:

<img width="582" height="143" alt="Image" src="https://github.com/user-attachments/assets/85cf98ca-668b-4736-8013-ac24eb14d191" />

The arrow points up when it's sorted ascending (and the tooltip indicates the action: "sort descending"). The PR:

<img width="525" height="160" alt="Image" src="https://github.com/user-attachments/assets/11035c3e-0a7b-451a-82d5-c8b014313c0e" />

In the default theme function:

<img width="560" height="216" alt="Image" src="https://github.com/user-attachments/assets/fd739bf0-a98c-4f7d-81c4-e30c5cfe5ef8" />

So there's consistency with the default theme function and also some continuity with what is already there for Seven. We'll see if people agree. Or if most people prefer to go back to the simple carets like the default then we'd remove the override in Seven altogether?

**Comment by indigoxela** [comment]
@herbdool I really appreciate your attempt to revive and quick-fix this issue, but I'm not sure if we're really done, yet. The initial report wasn't about the icon image type, but about its potential to confuse. I'd at least wait for feedback from @robertgarrigos, whether the approach to switch to SVG icons here and open a new one for wording is OK.

Most of the discussion was about confusion, too. 🤔

**Comment by robertgarrigos** [comment]
These icons represent, without confusion, for me, the actual order of the listed items in a better way.

**Comment by jenlampton** [comment]
I have opened https://github.com/backdrop/backdrop-issues/issues/6985 where we can work on switching the icons if we want, since this issue effectively switched from PNG to SVG, but the icons are similar.

**Comment by quicksketch** [comment]
I went ahead and merged this PR as-is. It effectively does NOT change the current status quo at all, instead it only swaps from PNG to SVG. We can continue the discussion about swapping the arrow direction in https://github.com/backdrop/backdrop-issues/issues/6985.
